src/dotnet/APIView/APIViewWeb/Client/src/shared/signalr.ts

```diff
@@ -58,8 +53,270 @@ $(() => {
+  reviewLeft.removeClass("border");
+  reviewLeft.removeClass("rounded-1");
+
+  let reviewRight = $("#review-right");
+  reviewRight.addClass("review-approved");
+  reviewRight.removeClass("border");
+  reviewRight.removeClass("rounded-1");
+}
+
+/**
+ * change the review panel border state from green "approved" to grey "not approved"
+ */
+function removeApprovalBorder() {
+    let $reviewLeft = $("#review-left");
+    $reviewLeft.removeClass("review-approved");
+    $reviewLeft.addClass("border");
+    $reviewLeft.addClass("rounded-1");
+
+    let $reviewRight = $("#review-right");
+    $reviewRight.removeClass("review-approved");
+    $reviewRight.addClass("border");
+    $reviewRight.addClass("rounded-1");
+}
+
+/**
+ * parse the approval spans for its existence and order
+ * @param $approvalSpans may contain <upper text>, <approve button>, and/or <lower text>
+ * @param approvedByText string for <lower text> that indicates preexisting approvers
+ * @param approvalPendingText string for <lower text> that indicates no current approvers
+ * @param approvesCurrentRevisionText string for <upper text> that indicates the current user did not approve
+ * @returns a dictionary with the index of the upper and lower text elements. Value is -1 if an element does not exist.
+ */
+function parseApprovalSpanIndex($approvalSpans: JQuery<HTMLElement>, approvedByText: string, approvalPendingText: string, approvesCurrentRevisionText: string) {
+  let indexResult = {
+      "approvers": -1,
+      "upperText": -1,
+  };
+
+  for (var i = 0; i < $approvalSpans.length; i++) {
+    let content = $approvalSpans[i].textContent;
+
+    if (!content) {
+      return indexResult;
+    }
+
+    if (content.includes(approvedByText) || content.includes(approvalPendingText)) {
+          indexResult["approvers"] = i;
+      }
+    if (content.includes(approvesCurrentRevisionText)) {
+      indexResult["upperText"] = i;
+      }
+  }
+
+  return indexResult;
+}
+
+/**
+ * call when the current user approves the current review. removes the upper text 
+ * @param upperTextIndex index of the upper text in @$approvalSpans
+ * @param $approvalSpans span that includes revision approval block 
+ */
+function removeUpperTextSpan(upperTextIndex: number, $approvalSpans: JQuery<HTMLElement>) {
+    if (upperTextIndex !== -1) {
+        let upperTextSpan: HTMLElement = $approvalSpans[upperTextIndex];
+      upperTextSpan.remove();
+    }
+}
+
+/**
+ * remove the @approver from list of @approvers 
+ * @param approvers list of preexisting approvers
+ * @param approver GitHub username of user to remove from the list 
+ */
+function removeApproverFromApproversList(approvers, approver) {
+  for (var i = 0; i < approvers.length; i++) {
+    if (approvers[i].innerHTML === approver) {
+      if (i === 0) {
+        approvers[i].nextSibling?.remove();
+      } else {
+        approvers[i].previousSibling?.remove();
+      }
+      approvers[i].remove();
+      break;
+    }
+  }
+}
+
```

Comment:
I think you can move these functions to review.module.ts since they are operating on elements in the review page.